### ISSUE TITLE
Custom CSS is not reflected when dark mode is selected for some mermaid configurations.

### ISSUE THREAD

**Issue by btbf** [squidfunk/mkdocs-material#3665]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I am using `flowchart BT` on mermaid and the CSS is not reflected when I set it to dark mode.
In the case of graph LR, it switches.

### Expected behaviour

I want to apply the style specified in extra.css.

### Actual behaviour

![marmaid](https://user-images.githubusercontent.com/37224549/156763787-37d91b55-ab06-4b75-afb1-55d8aa7c9f59.png)

```
mermaid
    flowchart BT
        a2[リレー1] & a3[リレー2] <-- 開放FW --> a4[カルダノネットワーク]
        subgraph ide1[プール]
            subgraph リレーIP指定FW
                a1[BP]
            end
                a1[BP] <--> a2[リレー1] & a3[リレー2]
        end
        c1[PC] --> ide1
        c1[PC] --> エアギャップ

```

```
 mermaid
    graph LR
        A[エアギャップ] -->|tx.signed| B[BP];
```

### Steps to reproduce

N/A

### Package versions

- Python: `python --version`
 Python 3.8
- MkDocs: `mkdocs --version`
 mkdocs, version 1.2.3
- Material: `pip show mkdocs-material | grep -E ^Version`
Version: 8.2.4+insiders.4.10.1

### Configuration

```yaml
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
extra_css:
  - stylesheets/extra.css
```

### System information

- Operating system: ... Windows11
- Browser: ...Chrome

**Comment by squidfunk** [comment]
Thanks for reporting. When reporting something like this please always share the Markdown you are using to render what is seen in the screenshot. Otherwise, I'm not able to reproduce it.

@pawamoy is this related to #3657?

**Comment by btbf** [comment]
Added markdown code.

**Comment by btbf** [comment]
![image](https://user-images.githubusercontent.com/37224549/156767881-0684da78-73e7-4ed3-8b8e-d0f9c7eec15f.png)

**Comment by btbf** [comment]
![image](https://user-images.githubusercontent.com/37224549/156767996-1b05a1a4-b89b-49fc-ac6c-a6fdb0eb90fa.png)

**Comment by pawamoy** [comment]
@squidfunk that definitely looks like the same issue! Flowchart clusters labels.

**Comment by squidfunk** [comment]
Fixed by @pawamoy in #3657. Thanks!

**Comment by squidfunk** [comment]
Released as part of 8.2.5.

**Comment by btbf** [comment]
My server and github did not improve after the upgrade.

**Comment by pawamoy** [comment]
@btbf are you trying to apply custom styling to Mermaid diagrams through that `extra.css` file? Can we see its contents?

**Comment by btbf** [comment]
> @btbf are you trying to apply custom styling to Mermaid diagrams through that `extra.css` file? Can we see its contents?

Yes,

http://49.12.225.142:8000/setup/0-start-guide/

**Comment by pawamoy** [comment]
I mean, what's in your `stylesheets/extra.css` file? Try adding `!important` to your CSS rules? :open_hands:

**Comment by btbf** [comment]
http://49.12.225.142:8000/stylesheets/extra.css

```
.label text{
    fill: :var(--md-mermaid-font-color) !important;
}
```
Is this not good enough?

**Comment by pawamoy** [comment]
I'm no expert but I believe it should be `var` and not `:var`.

**Comment by btbf** [comment]
I fixed it, but it didn't change.

**Comment by pawamoy** [comment]
Have a look at https://github.com/squidfunk/mkdocs-material/discussions/3618 and https://github.com/squidfunk/mkdocs-material/pull/3657
@squidfunk explains how to update the Mermaid styling. I won't be able to help further!

**Comment by btbf** [comment]
This phenomenon occurs when you specify the following in mkdocs.yml.
```
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

In the following cases, css is applied normally, but js is not applied. 
```
format: !!python/name:pymdownx.superfences.fence_div_format
```

**Comment by btbf** [comment]
This issue is not yet closed

**Comment by squidfunk** [comment]
You cannot apply custom styles via `extra_css` in `mkdocs.yml` to override styles for specific nodes in Mermaid.js. If you want to do that, you would have to fork the theme, because the CSS for Mermaid.js needs to be provided explicitly when Mermaid.js is initializing. It __must__ be inlined into the JavaScript, or Mermaid.js will not work locally when viewed via `file://` (which we fixed in #3560) .

If you fork the theme, you can add adjust the styles [in this file](https://github.com/squidfunk/mkdocs-material/blob/master/src/assets/javascripts/components/content/code/mermaid/index.css), but you will need to [rebuild/compile the theme](https://squidfunk.github.io/mkdocs-material/customization/#theme-development).
